tools/settings.py

```diff
@@ -207,11 +216,10 @@ def dict(self):
     return self.attrs
 
   def external_dict(self, skip_keys={}): # noqa
-    external_settings = {k: v for k, v in self.dict().items() if k not in INTERNAL_SETTINGS and k not in skip_keys}
-    # Only the names of the legacy settings are used by the JS compiler
-    # so we can reduce the size of serialized json by simplifying this
-    # otherwise complex value.
-    external_settings['LEGACY_SETTINGS'] = [l[0] for l in external_settings['LEGACY_SETTINGS']]
+    external_settings = {}
+    for key, value in self.dict().items():
+      if ((key in self.legacy_settings and value) or value != self.defaults.get(key)) and key not in INTERNAL_SETTINGS and key not in skip_keys:
```

Comment:
perhaps add a comment about legacy settings, or keep the old one? otherwise it seems odd why they are mentioned here.

also, why `and value` for legacy settings - do we not need to send falsey values too?

Reply:
Good point, I expanded this add added a comment.